### PR TITLE
[qa] Added qa-checks #111

### LINKED ISSUE #111: [qa] Add run-qa-check script
Add run-qa-check script, make sure it includes the 404 link check we have in openwisp-website and openwisp-radius, eg: https://github.com/openwisp/openwisp-radius/blob/master/run-qa-checks#L10-L16

### PR DESCRIPTION
**Pull request by devkapilbansal**
Fixes #111

Although I try to solve all errors, one error still exists in `_templates/layout.html`

errorfile

### CODE REVIEW

File: run-qa-checks [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+set -e
+
+echo ''
+echo 'Checking documentation build status'
+./runsphinx-build
+
+# check for broken link
+python -m http.server 8001 &> /dev/null & pid=$!
```
Comment (nemesisdesign) [inline run-qa-checks:10]: I think you should make sure only the build directory of sphinx is served, otherwise the crawler will try to browse everything

Comment (devkapilbansal) [reply run-qa-checks:10]: Okay thanks

**Comment by devkapilbansal** [comment]
@nemesisdesign Blank endline check failed. Should I skip checkendline from run-qa-checks or is there a way to not run those checks on build directory?

Probably, running openwisp-qa-check before sphinx check can solve this issue.

File: run-qa-checks [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+set -e
+
+echo ''
+echo 'Checking documentation build status'
+./runsphinx-build
```
Comment (nemesisdesign) [inline run-qa-checks:7]: run this and the broken link checks as last

**Review by nemesisdesign — COMMENTED** [review]
> @nemesisdesign Blank endline check failed. Should I skip checkendline from run-qa-checks or is there a way to not run those checks on build directory?

I don't think there's a way to indicate directories to ignore right now.

> Probably, running openwisp-qa-check before sphinx check can solve this issue.

yes, sphinx checks last-

> Also, in future with the release of the new version of openwisp-utils, we might need to skip **checkrst** too. Should an issue be opened for that?

Yes, you can proceed and open an issue for that.

Thanks!

File: run-qa-checks (hunk 2 above)
Comment (devkapilbansal) [reply run-qa-checks:16]: Okay thanks. I am adding a comment to make it clear why openwisp-utils checks are performed first

File: general/press.rst [hunk 3]
```diff
@@ -188,13 +188,9 @@ Logos and Graphic material
...
 </assets/design/openwisp-logo.svg>`
 
-.. raw:: html
-
-  <div align="center" class="align-center">
-    <img alt="../_downloads/openwisp-logo.svg"
-         src="../_downloads/openwisp-logo.svg"
-         style="background:#000;padding:25px;margin-bottom:24px" />
-  </div>
+.. image:: /assets/design/openwisp.org.svg
```
Comment (nemesisdesign) [inline general/press.rst:191]: this points to a different logo, moreover, we can't let go of this if we want the white logo to show properly, we need a black background.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]

**Comment by devkapilbansal** [comment]
@atb00ker review it too.

**Review by atb00ker — APPROVED** [review]
LGTM! :smile:

Please rebase with master.

File: general/press.rst [hunk 4]
```diff
@@ -191,8 +191,8 @@ Logos and Graphic material
 .. raw:: html
 
   <div align="center" class="align-center">
-    <img alt="../_downloads/openwisp-logo.svg"
-         src="../_downloads/openwisp-logo.svg"
+    <img alt="../../assets/design/openwisp-logo.svg"
+         src="../../assets/design/openwisp-logo.svg"
```
Comment (nemesisdesign) [inline general/press.rst:195]: if I delete the build and static assets dir and recreate it, I can't see the logo.
Can I know why are you changing this bit? I don't understand what problem was causing and why we need to change it.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]

File: general/press.rst (hunk 4 above)
Comment (devkapilbansal) [reply general/press.rst:195]: I change it because the older link was not working and pylink was showing that link as broken.

Comment (nemesisdesign) [reply general/press.rst:195]: Thanks for explaining. If you delete the build and static directories, then recreate the documentation with `make html`, can you see the logo? I see this:

Screenshot from 2020-10-04 16-09-55

Comment (nemesisdesign) [reply general/press.rst:195]: PS: the link works though, it opens the right logo. It's the src attribute of the image which does not seem to work.

File: general/press.rst [hunk 5]
```diff
@@ -191,8 +191,8 @@ Logos and Graphic material
 .. raw:: html
 
   <div align="center" class="align-center">
-    <img alt="../_downloads/openwisp-logo.svg"
-         src="../_downloads/openwisp-logo.svg"
+    <img alt="../../assets/design/openwisp-logo.svg"
```
Comment (nemesisdesign) [inline general/press.rst:194]: also notice that we use the image path as the `alt` attribute (alternative text), looks like a mistake, we can change it to `openwisp`

File: general/press.rst (hunk 4 above)
Comment (devkapilbansal) [reply general/press.rst:195]: It was error with path. It seems to work in starting when I tried but don't know why it fails, although corrected it now. Else shouldn't we use the path as `https://github.com/openwisp/openwisp2-docs/blob/master/assets/design/openwisp-logo.svg`.

File: general/press.rst (hunk 5 above)
Comment (devkapilbansal) [reply general/press.rst:194]: Oh I thought that it was intentional. I added alt attribute it other two logos too. But, should be use alt as openwisp for all three logos or I should proceed like `Openwisp Black Logo` in alt

**Review by nemesisdesign — APPROVED** [review]
Thanks!

**Pull request merged**